linked_list/linked_list.py

```diff
@@ -1,97 +1,197 @@
-
 # Defines a node in the singly linked list
 class Node:
 
-    def __init__(self, value, next_node = None):
+    def __init__(self, value, next_node=None):
         self.value = value
         self.next = next_node
 
+
+# composition relationship to class Node
 # Defines the singly linked list
 class LinkedList:
     def __init__(self):
-      self.head = None # keep the head private. Not accessible outside this class
+        self.head = None  # keep the head private. Not accessible outside this class
 
     # returns the value in the first node
     # returns None if the list is empty
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: constant, no traversing needed
+    # Space Complexity: constant, no new data structures
     def get_first(self):
-        pass
-
+        if self.head is None:
+            return None
+        return self.head.value
 
     # method to add a new node with the specific data value in the linked list
     # insert the new node at the beginning of the linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: constant
+    # Space Complexity: constant
     def add_first(self, value):
-        pass
+        new_node = Node(value)
+        new_node.next = self.head  # assign current head of linked list to the
+        self.head = new_node  # assign new node to head
 
     # method to find if the linked list contains a node with specified value
     # returns true if found, false otherwise
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: linear
+    # Space Complexity: constant
     def search(self, value):
-        pass
+        if self.head is None:
+            return False
+        current_node = self.head
+        target_node = Node(value)

```

Comment:
You can create a new node for the target, but it's not necessary. See my suggestion on line 45 ⬇️